Use new UpdatePolicyRoomName api command

neil-marcellini opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->
cc @marcaaron
### Details
<!-- Explanation of the change or anything fishy that is going on -->
Use the refactored UpdatePolicyRoomName api command, add offline feedback, and add RBR indicators. The RBR indicators probably won't be used because it is very unlikely that there will be errors from this command since there is front end validation.

### Fixed Issues
<!---
Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing.
Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the issue; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/Expensify/issues/212892

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
**A. Basic rename**

1. Sign in with any account
2. Create a workspace from the green plus (if needed)
3. Create a workspace room from the green plus (if needed)
4. Click the header of the workspace room to open the room details page
5. Click "Settings"
6. Rename the room and hit save
7. Verify that the room name updates correctly

**B. Offline tests**

1. Go offline
2. Update the room name again with a different name
3. Verify that the input is grayed out
4. Go back online
5. Verify that the input is no longer grayed out

**C. RBR tests**

1. Comment out these lines so that you can enter a duplicate room name
2. https://github.com/Expensify/App/blob/7bdd021cc6b3c91bcc448f34cca98450f5becd90/src/pages/ReportSettingsPage.js#L167-L169
3. Create another room with a different name
4. Click the header of the workspace room to open the room details page
5. Click "Settings"
6. Rename the room to a room name that already exists and hit save
7. Verify that a red dot, error message, and X button appear under the room name
8. Go back from settings and verify that a red dot shows on the settings menu item
9. Close the room details
10. Verify that there is a red dot on the room header and and in the LHN.
11. Open the room settings again
12. Dismiss the error by clicking the X
13. Verify that the room name reverts to the previous name and that all the red dots no longer show

- [ ] Verify that no errors appear in the JS console

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [x] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
Tests **A** and **B** from above.
- [ ] Verify that no errors appear in the JS console

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->
Offline
<img src="https://user-images.githubusercontent.com/26260477/184248785-a20c08a9-2c0e-4f7c-bb8e-214a169e767a.png" />


RBR
<img src="https://user-images.githubusercontent.com/26260477/184246828-cd6a4969-afde-4fa0-a451-7be2aa81fb92.png"/>
<img src="https://user-images.githubusercontent.com/26260477/184246948-02ba1029-a6a3-4f71-806a-e61f06379e35.png" width="300" />
<img src="https://user-images.githubusercontent.com/26260477/184247938-5179c7e1-e4dd-444f-8aa2-208625b9a86a.png" width="300" />


#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->

#### Android
<!-- Insert screenshots of your changes on the Android platform-->

neil-marcellini committed:
Create updatePolicyRoomName for api refactor

neil-marcellini committed:
Use reportName param for WAF

neil-marcellini committed:
Use new updatePolicyRoomName command

marcaaron commented on src/libs/actions/Report.js:
```diff
@@ -1466,6 +1466,52 @@ function renameReport(reportID, reportName) {
+ * Update the policy room name.
+ *
+ * @param {Object} policyRoomReport
+ * @param {*} reportName The updated reportName
```
```suggestion
 * @param {String} reportName The updated reportName
```
?

neil-marcellini replied:
Woops. Thanks for pointing that out.

marcaaron commented on src/libs/actions/Report.js:
```diff
@@ -1466,6 +1466,52 @@ function renameReport(reportID, reportName) {
+            value: {
+                reportName,
+                pendingFields: {
+                    reportName: 'update',
```
Let's create a CONST for `update` if there is not one already.

neil-marcellini replied:
Oh yeah, I added constants for that last week. Updated.

neil-marcellini committed:
Add offline feedback / errors for policy room name

neil-marcellini committed:
Merge branch 'main' into neil-UpdatePolicyRoomName

neil-marcellini committed:
Use rbr pending action update constant

neil-marcellini committed:
Fix JSDoc param type

neil-marcellini commented:
I'm holding on https://github.com/Expensify/App/pull/10317 so I can use the util function `getBrickRoadIndicatorStatusForReport`. I'll help review that and get it merged.

neil-marcellini committed:
Pass policyRoomName to UpdatePolicyRoomName

marcaaron commented:
Just a heads up that https://github.com/Expensify/App/pull/10317 is merged now

neil-marcellini commented:
Yep, thanks for getting that merged in. I will try to finish this off soon.

neil-marcellini committed:
Merge branch 'main' into neil-UpdatePolicyRoomName

neil-marcellini committed:
Check report errors and errorFields also

neil-marcellini committed:
Add brick road indicator to header view

neil-marcellini committed:
Create hasReportNameError

neil-marcellini committed:
Use hasReportNameError in HeaderView

neil-marcellini committed:
Fix hasReportNameError

neil-marcellini committed:
Allow empty string brickRoadIndicator

neil-marcellini committed:
Brick road indicator on report details settings

neil-marcellini commented on src/components/MenuItem.js:
```diff
@@ -132,11 +132,11 @@ const MenuItem = props => (
                         </View>
                     )}
-                    {props.brickRoadIndicator && (
+                    {Boolean(props.brickRoadIndicator) && (
```
Treat this as a boolean in case it's an empty string.

neil-marcellini commented on src/components/menuItemPropTypes.js:
```diff
@@ -67,9 +67,9 @@ const propTypes = {
-    brickRoadIndicator: PropTypes.oneOf(['error']),
+    /** If we need to show a brick road indicator or not. For now only value allowed is 'error' or '',
+     * but we will add 'success' later for manual requests */
+    brickRoadIndicator: PropTypes.oneOf([CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR, '']),
```
The empty string is the empty value so that functions that get the brick road indicator property can always return a string.

neil-marcellini committed:
Use empty string for empty brickRoadIndicator

neil-marcellini committed:
Allow the error message to use the full row

neil-marcellini commented:
Here's an alternative option for RBR / Offline feedback for the room name input. Although I like how the save button isn't grayed out while offline I think the error message looks bad. However, the error would be very rare.
![image](https://user-images.githubusercontent.com/26260477/184246807-fbf03bf1-717c-4d71-9b40-456b6b5863b2.png)
<img src="https://user-images.githubusercontent.com/26260477/184246966-8589ba00-3c46-4bcb-ba30-3fb32aedf3fa.png" width="300" />

neil-marcellini committed:
Name the onClose callback for what it does

neil-marcellini commented on src/components/RoomNameInput.js:
```diff
@@ -91,6 +95,7 @@ class RoomNameInput extends Component {
     render() {
         return (
             <TextInput
+                ref={this.props.forwardedRef}
```
I added a ref so the input value can be cleared from the parent.

neil-marcellini committed:
Remove redundant comments

neil-marcellini committed:
Remove deprecated rename report

neil-marcellini committed:
Remove policy room renamed growl

neil-marcellini committed:
Rename to match api command name

neil-marcellini committed:
Merge branch 'main' into neil-UpdatePolicyRoomName

marcaaron commented on src/pages/ReportSettingsPage.js:
```diff
@@ -107,23 +108,46 @@ class ReportSettingsPage extends Component {
+     */
+    setRoomNameInputRef(el) {
+        this.roomNameInputRef = el;
+    }
```
Can we get rid of this function and just do it inline?

```js
<Component
    ref={el => this.roomNameInputRef = el}
```

neil-marcellini replied:
Yeah. That looks like the way we usually do it. I'll keep that in mind for the future. I was looking at ReportActionCompose which doesn't set the ref inline.

marcaaron commented on src/pages/ReportSettingsPage.js:
```diff
@@ -107,23 +108,46 @@ class ReportSettingsPage extends Component {
+
+        // Reset the input's value back to the previously saved report name
+        if (this.roomNameInputRef) {
+            this.roomNameInputRef.value = this.props.report.reportName.replace(CONST.POLICY.ROOM_PREFIX, '');
```
I wasn't sure about this part at first, but then remembered React Native inputs are more performant when they are uncontrolled.

Still, I'm not sure if modifying the value like this is 100% correct - I vaguely recall there being issues on native related to assigning the value directly. Did we test this out there?

neil-marcellini replied:
I will test on iOS and Android to see if it works there. I did a quick google search for how to clear a text input and went off of [this SO](https://stackoverflow.com/questions/45249807/clear-react-native-textinput), so it seems like the accepted way to do it.

neil-marcellini replied:
Woops, that SO is about clearing the text input, which isn't exactly what I want to do in this case. You're right that this approach doesn't work. When I tested on iOS I was unable to enter anything after the value was directly set.

Should I use [setNativeProps](https://reactnative.dev/docs/direct-manipulation#setnativeprops-to-edit-textinput-value) or should I make this a controlled input, or something else?

neil-marcellini replied:
Looks like we use `setNativeProps` in a couple other places, and it works well, so I'll go with that.
https://github.com/Expensify/App/blob/782ea4a7c1350b36519bf4dff2512e0fa37b07e8/src/pages/home/report/ReportActionCompose.js#L375
https://github.com/Expensify/App/blob/1bfae69c3dce03d017270cc9838c480610e925c1/src/components/Form.js#L167

marcaaron replied:
Controlled components have some performance issues on native so I usually recommend the uncontrolled. But if this were a web only app I'd say to use controlled components as I prefer working with them (e.g. clearing an input just requires the parent setting a state value to `''` which is more intuitive IMO). `setNativeProps` looks like the only option to me.

marcaaron commented on src/pages/home/HeaderView.js:
```diff
@@ -146,6 +149,16 @@ const HeaderView = (props) => {
+                                        height={variables.iconSizeSmall}
+                                        width={variables.iconSizeSmall}
+                                    />
+                                </View>
```
NAB, I think it's probably time to make this a re-usable component.

neil-marcellini replied:
I would be happy to do that in a follow up.

marcaaron commented on src/libs/actions/Report.js:
```diff
@@ -1377,30 +1377,61 @@ function createPolicyRoom(policyID, reportName, visibility) {
+            value: {
+                pendingFields: {
+                    reportName: null,
+                },
```
Should we also clear any `reportName` errors on success?

neil-marcellini replied:
Yeah good point, we should clear the errors. I think it would actually be better in the optimistic data. If I update with an existing name and I see an error, and then I go to update with a different name, I would want to see the error disappear right away.

marcaaron replied:
Yeah, let's go with that 👍

marcaaron commented on src/components/MenuItem.js:
```diff
@@ -42,7 +42,7 @@ const defaultProps = {
     interactive: true,
     fallbackIcon: Expensicons.FallbackAvatar,
-    brickRoadIndicator: undefined,
+    brickRoadIndicator: null,
```
Should this be an `''` as we said in `propTypes` that it would be `'error'` or `''`?

neil-marcellini replied:
Ah yep, good catch.

marcaaron reviewed (changes requested):
Great job all around 👍👍 

Mostly had NAB comments and a few questions.

neil-marcellini commented on src/components/menuItemPropTypes.js:
```diff
@@ -67,9 +67,9 @@ const propTypes = {
-     * but we will add `success` later for manual requests */
-    brickRoadIndicator: PropTypes.oneOf(['error']),
+    /** If we need to show a brick road indicator or not. For now only value allowed is 'error' or '',
+     * but we will add 'success' later for manual requests */
```
True, I will update it.

neil-marcellini committed:
Simplify brickRoadIndicator comment

neil-marcellini commented on src/libs/OptionsListUtils.js:
```diff
@@ -191,18 +191,26 @@ function hasReportDraftComment(report, reportsWithDraft = {}) {
  * @returns {String}
  */
 function getBrickRoadIndicatorStatusForReport(report, reportActions) {
+    const errors = lodashGet(report, 'errors', {});
```
Good idea to distinguish these from the reportActionErrors.

neil-marcellini committed:
Rename error vars that are specific to the report

neil-marcellini commented on src/libs/actions/Report.js:
```diff
@@ -1377,30 +1377,61 @@ function createPolicyRoom(policyID, reportName, visibility) {
+        },
+        pendingFields: {
+            reportName: null,
+        },
```
Yep, that's exactly it.

neil-marcellini committed:
Add a key to menu items for brick road indicator

neil-marcellini committed:
Remove unused props

neil-marcellini commented on src/pages/ReportDetailsPage.js:
```diff
@@ -156,6 +157,12 @@ class ReportDetailsPage extends Component {
                         const keyTitle = item.translationKey ? this.props.translate(item.translationKey) : item.title;
+                        const brickRoadIndicator = (
+                            ReportUtils.hasReportNameError(this.props.report)
+                            && item.translationKey === 'common.settings'
```
Yeah, that felt a bit hacky. I added a key to the menu items which simplifies things a bit.

neil-marcellini committed:
Make default brickRoadIndicator match prop type

neil-marcellini committed:
Clear reportName errors optimistically

neil-marcellini committed:
Set roomNameInputRef inline for simplicity

neil-marcellini committed:
Fix resetting the room name input on native

Revert accidentally commenting out validation

Fix resetting the room name input on native

neil-marcellini committed:
Use minimum vertical space for RBR error text

neil-marcellini commented:
The errors were not looking good on iOS because the error text was taking up the full height due to `flex1` and therefore it was not centered.

**Before**
<img src="https://user-images.githubusercontent.com/26260477/184449639-91614390-cf01-4bbc-aea2-8b4f7e036496.png" width="300" />
**After**
<img src="https://user-images.githubusercontent.com/26260477/184449649-7b662a92-ee47-4927-9036-acbac8693c55.png" width="300" />

**Before**
<img src="https://user-images.githubusercontent.com/26260477/184449797-a95e1eb3-e138-42e8-a15d-b0f11695076f.png" width="300" />

**After**
<img src="https://user-images.githubusercontent.com/26260477/184449836-2ee7d259-c725-4395-b27b-a7335bea3c0b.png" width="300" />

neil-marcellini commented:
Ok, ready for another review please! Also, cc @Expensify/design to make sure it looks good.

marcochavezf commented on src/pages/ReportDetailsPage.js:
```diff
@@ -81,6 +83,7 @@ class ReportDetailsPage extends Component {
 
         if (ReportUtils.isPolicyExpenseChat(this.props.report) || ReportUtils.isChatRoom(this.props.report)) {
             this.menuItems.push({
+                key: 'settings',
```
I think we should move these string values to CONST.js, no?

neil-marcellini replied:
Sure, done.

marcochavezf commented on src/libs/actions/Report.js:
```diff
@@ -1377,30 +1377,64 @@ function createPolicyRoom(policyID, reportName, visibility) {
- * Renames a user created Policy Room.
- * @param {String} reportID
- * @param {String} reportName
+ * @param {Object} policyRoomReport
```
```suggestion
 * @param {Object} policyRoomReport
 * @param {String} policyRoomReport.reportID
 * @param {String} policyRoomReport.reportName
```

marcochavezf reviewed (commented):
LGTM! Just a few comments 👍🏽

neil-marcellini committed:
Add accessed report fields to JSDoc

neil-marcellini committed:
Use constants for report details menu items

neil-marcellini commented:
Ready for another round of reviews.

marcaaron commented on src/pages/ReportDetailsPage.js:
```diff
@@ -155,17 +160,21 @@ class ReportDetailsPage extends Component {
-                        const keyTitle = item.translationKey ? this.props.translate(item.translationKey) : item.title;
+                        const brickRoadIndicator = (
+                            ReportUtils.hasReportNameError(this.props.report)
+                            && item.key === 'settings'
```
```suggestion
                            && item.key === CONST.REPORT_DETAILS_MENU_ITEM.SETTINGS
```

neil-marcellini replied:
Woops, thanks for catching that.

marcaaron reviewed (changes requested):
Looking good. Spotted a small improvement.

neil-marcellini committed:
Use constant

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

marcaaron commented:
Oh hmm did this need to HOLD on the production web deploy? The command we are calling doesn't exist yet 🙃

marcaaron commented:
Looks like it is still on staging (we are probably OK as long as the production deploy goes out before the App deploy) but probably should have been on HOLD.

neil-marcellini commented:
Yeah true. I think it's fairly likely that the Web deploy will be done first. Is there anything I can do to make sure that happens?

marcaaron commented:
I think the easiest would be to put a deploy blocker label on this PR or the linked issue when it goes to staging. Not sure about the timing on that. In theory, QA should test this and it should fail QA if Web-Expensify is not deployed to production.

francoisl commented:
https://github.com/Expensify/Web-Expensify/pull/34501 just hit production, going to remove the deploy blocker label.

neil-marcellini commented:
Ha sorry I was just about to say that.
